FamliCareClientProject/famlicare-client-project
# Enhance CareVault Module with Improved Logging and Error Handling

## Description
This PR enhances the CareVault module with improved logging and error handling across the saga, component, and router. These changes will provide better visibility into application behavior and make debugging easier, especially when using Heroku's Sentry and PaperTrail plugins.

## Changes
- careVault.saga.js: Added detailed logging for each saga function, implemented consistent error handling, and improved payload logging.
- CareVault.jsx: Added file size check before upload, enhanced logging for user actions, and improved error messages and user feedback.
- careVault.router.js: Replaced console.log with a logger module, added detailed logging for all route handlers, and improved error responses with more context.

## Why
These changes will significantly improve our ability to monitor and debug the CareVault module, especially in the Heroku environment. The enhanced logging will provide more context in PaperTrail logs, while the improved error handling will give more detailed information in Sentry error reports.

## How to Test
1. Deploy to a staging environment on Heroku.
2. Attempt to upload files of various sizes and types.
3. Verify that appropriate logs are appearing in PaperTrail.
4. Intentionally cause errors (e.g., try to upload without a loved one associated) and check Sentry for detailed error reports.
5. Verify that all existing functionality still works as expected.

## Additional Notes
- Make sure that the logger module is correctly configured to work with PaperTrail on Heroku.
- Update any relevant documentation to reflect these changes, especially regarding new log messages and error handling.

## Screenshots (if applicable)
[Add any relevant screenshots here]
